pkg/provider/prefix/provider.go

```diff
@@ -372,22 +372,28 @@ func (p *ipv4PrefixProvider) GetPool(nodeName string) (pool.Pool, bool) {
 }
 
 func (p *ipv4PrefixProvider) IsInstanceSupported(instance ec2.EC2Instance) bool {
+	// if instance is non-Windows, don't send node event, simply return false
+	if instance.Os() != config.OSWindows {
+		return false
+	}
+
 	instanceName := instance.Name()
 	instanceType := instance.Type()
 	isNitroInstance, err := utils.IsNitroInstance(instanceType)
 
+	// if instance type is not found in vpc limits, send node event for instance type not supported for Windows
 	if errors.Is(err, utils.ErrNotFound) {
 		msg := fmt.Sprintf("The instance type %s is not supported for Windows", instanceType)
 		utils.SendNodeEvent(p.apiWrapper.K8sAPI, instanceName, "Unsupported", msg, v1.EventTypeWarning, p.log)
 		return false
 	}
 
-	if instance.Os() == config.OSWindows && err == nil && isNitroInstance {
+	if err == nil && isNitroInstance {
 		return true
 	}
 
-	// if instance is non-nitro or non-Windows, PD is not supported
-	msg := fmt.Sprintf("The instance type %s is not supported for prefix delegation", instanceType)
+	// if instance type is non-nitro, PD is not supported
+	msg := fmt.Sprintf("The instance type %s is not supported for Windows prefix delegation", instanceType)
 	utils.SendNodeEvent(p.apiWrapper.K8sAPI, instanceName, "Unsupported", msg, v1.EventTypeWarning, p.log)
```

Comment:
Can you use `utils.UnsupportedInstanceTypeReason`?

Reply:
Yeah sure, I've updated it. I haven't rebased with master branch so I fixed it manually. Hope that's ok.